src/hooks/BalanceSummary.test.tsx

```diff
@@ -0,0 +1,173 @@
+import React from 'react'
+import { render } from '../testUtils'
+import { act } from 'react-dom/test-utils'
+
+import { useBalanceSummary, WalletBalanceSummary } from './BalanceSummary'
+import { WalletInfo } from '../context/WalletContext'
+
+function setup(walletInfo: WalletInfo | null) {
+  const returnVal: unknown = {}
+  const TestComponent: React.FunctionComponent = () => {
+    const val = useBalanceSummary(walletInfo)
+    Object.assign(returnVal, val)
+    return <></>
+  }
+
+  render(<TestComponent />)
+
+  return returnVal
+}
+
+describe('BalanceSummary', () => {
+  it('should handle missing wallet info without errors', () => {
+    const data = {}
+
+    act(() => {
+      Object.assign(data, setup(null))
+    })
+
+    const balanceSummary = data as WalletBalanceSummary
+
+    expect(balanceSummary).toBeDefined()
+    expect(balanceSummary.totalBalance).toBeNull()
+    expect(balanceSummary.availableBalanceDontUseYet).toBeNull()
+    expect(balanceSummary.calculatedAvailableBalanceInSats).toBeNull()
+    expect(balanceSummary.calculatedFrozenOrLockedBalanceInSats).toBeNull()
+    expect(balanceSummary.accountBalances).toBeNull()
+  })
+
+  it('should handle new format', () => {
+    const data = {}
+
+    act(() => {
+      Object.assign(
+        data,
+        setup({
+          data: {
+            utxos: {
+              utxos: [],
+            },
+            display: {
+              walletinfo: {
+                wallet_name: 'test.jmdat',
+                total_balance: '1.00000001',
```

Comment:
> it('should handle new format', () => {

That's the old format as well right?

Reply:
> Cool thanks for adding tests. If you want to add more feel free to do so and just merge whenever you think it is ready to be merged.

Yes, I want to add more tests. It is not quite ready yet. Will propably merge tomorrow.

> That's the old format as well right?

Yes, copy paste error, thanks.